server/routes/api/chart.py

```diff
@@ -213,48 +210,74 @@ def choropleth_data(dcid):
     API Returns:
         Dictionary with
             key as stat var
-            value as object with date,dictionary of place: value, number of data points, exploreUrl, list of sources
+            value as object with date,dictionary of place: value, number of
+            data points, exploreUrl, list of sources
     """
-    all_stat_vars, choropleth_configs = get_choropleth_sv()
-    geos, _ = get_choropleth_places(dcid)
-    if not all_stat_vars or not geos:
+    configs = get_choropleth_configs()
+    stat_vars, denoms = get_stat_vars(configs)
+    display_dcid, display_level = get_choropleth_display_level(dcid)
+    geos = dc_service.get_places_in([display_dcid],
+                                    display_level).get(display_dcid, [])
+    if not stat_vars or not geos:
         return Response(json.dumps({}), 200, mimetype='application/json')
     # Get data for all the stat vars for every place we will need and process the data
-    all_sv_data = dc_service.get_stats_all(geos, list(all_stat_vars))
-    if not 'placeData' in all_sv_data:
-        return Response(json.dumps({}), 200, mimetype='application/json')
-    processed_data = process_choropleth_data(all_sv_data['placeData'])
-
+    sv_data = dc_service.get_stat_set_within_place(display_dcid, display_level,
+                                                   list(stat_vars),
+                                                   "").get('data', {})
+    denoms_data = {}
+    for denom in denoms:
```

Comment:
For denom data, we want data from multiple dates so that we can find the date that best matches the stat var. Therefore, I don't think we should use get_stat_set_within_place

Reply:
I see, and this is where the complexity comes from